### PR TITLE
add missing ncurses dep for Clang 5.0.x, 6.0.1, 7.0.x

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
node2645.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/d5a4d28ed437cef959d3265640768eb8 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 9 (7 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/8ea5ae84fe7b0ca5dfe2496930572aa1 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 9 (7 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/2b721ec208d9c84a454e90cd18c9d603 for a full test report.

**Comment by boegel** [comment]
@lexming `No space left on device` :)

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 9 out of 9 (7 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/2d0bea6e671b18b48aa32c5be2f88fc9 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 9 out of 9 (7 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/32baa2d79e9aa3274210fd866bfe79c1 for a full test report.

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-3.1.4-GCC-8.3.0.eb [hunk 1]
```diff
@@ -17,7 +17,9 @@ checksums = [
 
 dependencies = [
     ('zlib', '1.2.11'),
-    ('hwloc', '1.11.12')
+    ('hwloc', '1.11.12'),
+    ('UCX', '1.6.1'),
+    ('PMIx', '3.1.4'),
```
Comment (lexming) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-3.1.4-GCC-8.3.0.eb:22]: These changes to OpenMPI seem not related to this PR. Were those intended?

Comment (boegel) [reply easybuild/easyconfigs/o/OpenMPI/OpenMPI-3.1.4-GCC-8.3.0.eb]: No, not intended, fixed with force push, sorry...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node2603.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/c35f6df6345cbc7c8a17f5361ca196d2 for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @boegel !
